tests/functional/ctlplane/base_test.go

```diff
@@ -632,3 +675,95 @@ func SimulateControlplaneReady() {
 
 	}, timeout, interval).Should(Succeed())
 }
+
+// GetSampleTopologySpec - A sample (and opinionated) Topology Spec
+func GetSampleTopologySpec() map[string]interface{} {
+	// Build the topology Spec
+	topologySpec := map[string]interface{}{
+		"topologySpreadConstraints": []map[string]interface{}{
+			{
+				"maxSkew":           1,
+				"topologyKey":       k8s_corev1.LabelHostname,
+				"whenUnsatisfiable": "ScheduleAnyway",
+			},
+		},
+	}
+	return topologySpec
+}
+
+// CreateTopology - Creates a Topology CR based on the spec passed as input
+func CreateTopology(topology types.NamespacedName, spec map[string]interface{}) client.Object {
+	raw := map[string]interface{}{
+		"apiVersion": "topology.openstack.org/v1beta1",
+		"kind":       "Topology",
+		"metadata": map[string]interface{}{
+			"name":      topology.Name,
+			"namespace": topology.Namespace,
+		},
+		"spec": spec,
+	}
+	return th.CreateUnstructured(raw)
+}
+
+// GetGlance
+func GetGlance(name types.NamespacedName) *glancev1.Glance {
```

Comment:
In an ideal world these functions might be replaced (like we do for memcached) by the `api/tests` helpers provided by each service operator. This is something we need to reduce the amount of utility functions here.

Reply:
https://issues.redhat.com/browse/OSPRH-14532 to track this work